mailserver/db_key.go

```diff
@@ -0,0 +1,44 @@
+package mailserver
+
+import (
+	"encoding/binary"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	// DBKeyLength is a size of the envelope key.
+	DBKeyLength = common.HashLength + timestampLength
+)
+
+// DBKey key to be stored on db.
+type DBKey struct {
+	timestamp uint32
+	hash      common.Hash
+	raw       []byte
+}
+
+// Bytes returns a bytes representation of the DBKey.
+func (k *DBKey) Bytes() []byte {
+	return k.raw
+}
+
+// NewDBKey creates a new DBKey with the given values.
+func NewDBKey(t uint32, h common.Hash) *DBKey {
```

Comment:
```suggestion
func NewDBKey(timestamp uint32, h common.Hash) *DBKey {
```

of alias uint32 as `timestamp` type, improves readability

Reply:
or we can just accept `time.Time` and convert it to timestamp only for the raw implementation

Reply:
> of alias uint32 as timestamp type, improves readability

It's only seemingly better because saying this you know that `timestamp` is `uint32` :) When one would like to call `NewDBKey` later, they will need to figure out what `timestamp` is just to see it's `uint32`. We had this discussion and if there is no methods for a custom type, it's rarely worth to do aliasing.

> or we can just accept time.Time and convert it to timestamp only for the raw calculation

In most of the use cases, it will mean converting `int64` or `uint32` to `time.Time` and then back to `uint32` in `NewDBKey`.

I like changing `t` to `timestamp`. I think it solves readability without introducing other issues.